qiita_db/metadata_template/test/test_prep_template.py

```diff
@@ -1308,14 +1308,34 @@ def test_check_restrictions(self):
 
         del self.metadata['primer']
         pt = npt.assert_warns(QiitaDBWarning, PrepTemplate.create,
-                              self.metadata, self.new_raw_data,
-                              self.test_study, self.data_type)
+                              self.metadata, self.test_study, self.data_type)
 
         obs = pt.check_restrictions(
             [PREP_TEMPLATE_COLUMNS['EBI'],
              PREP_TEMPLATE_COLUMNS_TARGET_GENE['demultiplex']])
         self.assertEqual(obs, {'primer'})
 
+    def test_raw_data(self):
+        """Returns the raw_data associated with the prep template"""
+        self.assertEqual(self.tester.raw_data, 1)
+
+        pt = PrepTemplate.create(self.metadata, self.test_study,
+                                 self.data_type_id)
+        self.assertEqual(pt.raw_data, None)
+
+    def test_raw_data_setter_error(self):
+        rd = RawData(1)
+        with self.assertRaises(QiitaDBError):
+            self.tester.raw_data = rd
+
+    def test_raw_data_setter(self):
+        rd = RawData(1)
+        pt = PrepTemplate.create(self.metadata, self.test_study,
+                                 self.data_type_id)
+        self.assertEqual(pt.raw_data, None)
+        pt.raw_data = rd
+        self.assertEqual(pt.raw_data, 1)
```

Comment:
Note that the RawData id is hard-coded above: `rd = RawData(1)`. However, if this ever changes, by using rd.id it will simplify future changes, so changing. Thanks!